Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp

```diff
@@ -152,20 +153,57 @@ namespace AZ
             return m_visScene->GetEntryCount();
         }
 
+        CullingDebugContext& CullingScene::GetDebugContext()
+        {
+            return m_debugCtx;
+        }
+
+        const AzFramework::IVisibilityScene* CullingScene::GetVisibilityScene() const
+        {
+            return m_visScene;
+        }
+
+        static AzFramework::EntityContextId GetEntityContextIdForOcclusion(const AZ::RPI::Scene* scene)
+        {
+            auto sceneSystem = AzFramework::SceneSystemInterface::Get();
+            AZ_Assert(sceneSystem, "Attempting to retrieve the entity context ID for a scene before the scene system interface is ready.");
+
+            AzFramework::EntityContextId resultId = AzFramework::EntityContextId::CreateNull();
+            sceneSystem->IterateActiveScenes(
+                [&](const AZStd::shared_ptr<AzFramework::Scene>& azScene)
+                {
+                    AZ::RPI::ScenePtr* rpiScene = azScene->FindSubsystemInScene<AZ::RPI::ScenePtr>();
+                    if (rpiScene && (*rpiScene).get() == scene)
+                    {
+                        AzFramework::EntityContext** entityContext =
+                            azScene->FindSubsystemInScene<AzFramework::EntityContext::SceneStorageType>();
+                        if (entityContext)
+                        {
+                            const AzFramework::EntityContextId contextId = (*entityContext)->GetContextId();
+                            if (AzFramework::OcclusionRequestBus::FindFirstHandler(contextId))
```

Comment:
Nit: Could use HasHandlers(contextId) instead of FindFirstHandler for slightly better readability.